VHAR-3780 - triggerit, joilla päivitetään toteutuneet_kustannukset taulua

rckopteri opened the pull request:
Muokataan toteutuneet_kustannukset taulun sisältöä kun kustannusarvioitu_tyo tai johto_ja_hallintakorvaus tauluisat siirrettyjä datoja muokataan tai poistetaan

rckopteri committed:
Muokataan toteutuneet_kustannukset taulun sisältöä

kun kustannusarvioitu_tyo tai johto_ja_hallintakorvaus tauluisat siirrettyjä datoja muokataan tai poistetaan

jpluomajoki-solita commented:
Katseltiin PR läpi puhelussa, todettiin että kuluvana vuonna, mutta aiempana kuukautena muokatut budjetit eivät välttämättä peilaannu toteutuneet_kustannukset-taulussa.

Huomattiin myös bugi näkymättömästä tuntipalkasta ui:ssa, @rckopteri kirjaa bugin :imp:

jpluomajoki-solita commented:
Lisäksi migraation versionumero pitää päivittää

rckopteri committed:
Merge branch 'develop' into VHAR-3780-paivita-toteutuneet-kustannukset

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_858__.sql

rckopteri committed:
Tehdään Triggeri päivittyneelle budjetoidulle tiedolle,

joka pitää siirtää toteutuneet_kustannukset tauluun. Sekä proseduuri, joka hoitaa siirron

rckopteri committed:
Otetaan käyttöön uusi proceduuri, joka hoitaa budjetoitujen kustannusten siirron

toteutuneet_kustannukset tauluun

rckopteri committed:
Merge branch 'develop' into VHAR-3780-paivita-toteutuneet-kustannukset

jpluomajoki-solita commented on tietokanta/src/main/resources/db/migration/V1_860__.sql:
```diff
@@ -0,0 +1,237 @@
+-- Merkitään kustannusarvioitu_tyo ja johto_ja_hallintakorvaus tauluihin tieto, että onko
+-- kyseinen rivi jo siirretty toteutuneet_kustannukset tauluun.
+ALTER TABLE kustannusarvioitu_tyo
+    ADD COLUMN "siirrety?" BOOLEAN NOT NULL DEFAULT false;
```
Typo? siirrety -> siirretty

jpluomajoki-solita commented on tietokanta/src/main/resources/db/migration/V1_860__.sql:
```diff
@@ -0,0 +1,237 @@
+    ADD COLUMN rivin_tunnistin TEXT NOT NULL;
+
+-- Luodaan funktio, jonka update tai delete kustannusarvoitu_tyo trigger ajaa
+CREATE OR REPLACE FUNCTION paivita_toteutuneet_kustannukset_kat() RETURNS trigger AS
```
Eli tämä päivittää tai poistaa toteutuneen kustannuksen, mikäli kustannusarvioitua työtä muokataan. Miten jos muokataan kuluvan kuukauden kustannusarvioitua työtä, siirtyykö tämä silloin jo toteutuneisiin kustannuksiin? Mitä jos muokataan jo "lukitussa" kuukaudessa olevaa kustannusarviota?

jpluomajoki-solita replied:
Selvennys puhelusta: Muokataan vain niitä, joilla on toteutuneet-taulussa jo tunniste, eli kuluvan kuukauden toimenpiteiden päivitys ei siirrä niitä tänne.

jpluomajoki-solita reviewed (commented):
Kommentti ja kysymys vielä toiminnallisuuteen liittyen, muuten näyttää selkeältä.

jpluomajoki-solita reviewed (commented):
Puhelun jälkeen voidaan todeta, että näyttää hyvältä kirjoitusvirheen korjauksen jälkeen.

rckopteri committed:
Merge branch 'develop' into VHAR-3780-paivita-toteutuneet-kustannukset

rckopteri committed:
Merge branch 'VHAR-3780-paivita-toteutuneet-kustannukset' of github.com:finnishtransportagency/harja into VHAR-3780-paivita-toteutuneet-kustannukset

rckopteri committed:
Korjataan kolumnin nimestä typo
